### ISSUE TITLE
False positives for `prefer-to-be-undefined`

### ISSUE THREAD

**Issue by alexlrobertson** [jest-community/eslint-plugin-jest#28]
```jsx
test('it should work', () => {
    const foo = {};

    expect(foo).toEqual({});
});
```

I'm getting false positives for expectations that compare to object and array literals.
